File: app/src/main/java/com/work/lazxy/writeaway/mvpframe/concrete/planning/PlanningPresent.java
package com.work.lazxy.writeaway.mvpframe.concrete.planning;


import android.util.Log;

import com.work.lazxy.writeaway.entity.PlanningEntity;
import com.work.lazxy.writeaway.mvpframe.base.BaseObserver;

import java.util.List;

/**
 * Created by Lazxy on 2017/2/22.
 */

public class PlanningPresent extends PlanningContract.Presenter {
    @Override
    public void getPlanning() {
        observe(mModel.getPlanning(), new BaseObserver<List<PlanningEntity>>(mView) {
            @Override
            protected void onSuccess(List<PlanningEntity> plannings) {
                if(plannings!=null){
                    mView.setPlanning(plannings);
                }
            }
        });
    }

    @Override
    public void savePlanning(List<PlanningEntity> plannings) {
        observe(mModel.savePlanning(plannings), new BaseObserver<String>(mView) {
            @Override
            protected void onSuccess(String s) {
                Log.i("DEBUG","saved success");
            }
        });
    }

    @Override
    public void addPlanning(PlanningEntity planning) {
        observe(mModel.addPlanning(planning), new BaseObserver<String>() {
            @Override
            protected void onSuccess(String s) {
                Log.i("DEBUG","saved success");
            }
        });
    }
}

File: app/src/main/java/com/work/lazxy/writeaway/mvpframe/concrete/planning/PlanningPresent.kt
package com.work.lazxy.writeaway.mvpframe.concrete.planning

import android.util.Log
import com.work.lazxy.writeaway.entity.PlanningEntity
import com.work.lazxy.writeaway.mvpframe.base.BaseObserver

/**
 * Created by Lazxy on 2017/2/22.
 */
class PlanningPresent : PlanningContract.Presenter() {
    override fun getPlanning() {
        observe(mModel.planning, object : BaseObserver<List<PlanningEntity>>(mView) {
            override fun onSuccess(plannings: List<PlanningEntity>) {
                mView.setPlanning(plannings)
            }
        })
    }

    override fun savePlanning(plannings: List<PlanningEntity>) {
        observe(mModel.savePlanning(plannings), object : BaseObserver<String>(mView) {
            override fun onSuccess(s: String) {
                Log.i("DEBUG", "saved success")
            }
        })
    }

    override fun addPlanning(planning: PlanningEntity) {
        observe(mModel.addPlanning(planning), object : BaseObserver<String>() {
            override fun onSuccess(s: String) {
                Log.i("DEBUG", "saved success")
            }
        })
    }
}